Hazel/src/Hazel/ImGui/ImGuiConsoleSink.h

```diff
@@ -0,0 +1,62 @@
+#include "spdlog/sinks/base_sink.h"
+#include "ImGuiConsole.h"
+
+namespace Hazel {
+
+	template<class Mutex>
+	class ImGuiConsoleSink : public spdlog::sinks::base_sink<std::mutex>
+	{
+	public:
+		explicit ImGuiConsoleSink(bool forceFlush = false, uint8_t bufferCapacity = 10)
+			: m_MessageBufferCapacity(forceFlush ? 1 : bufferCapacity), m_MessageBuffer(std::vector<Ref<ImGuiConsole::Message>>(forceFlush ? 1 : bufferCapacity))
```

Comment:
When would you want this to buffer? This flushing mechanism seems unusual there is no disk io or anything, its just a queue of messages which will be shown to the user and displayed within the draw tick